src/ast/ProgramDecl.ts

```diff
@@ -0,0 +1,38 @@
+import {ModuleDecl} from "./ModuleDecl";
+import {DirDecl} from "./DirDecl";
+import {Tokenizer} from "../util/Tokenizer";
+
+/**
+ * Represents a TypeScript project to be generated.
+ *
+ * Child of DirDecl since a TypeScript project is a directory with a name and modules.
+ *
+ * e.g. ‘project ’ (str | STRLIST) MODULE? (DIR|CLASS|INTERFACE)* nl
+ */
+export class ProgramDecl extends DirDecl {
+    projectName: string;
```

Comment:
are we still planning on allowing for the user to specify multiple project names? (e.g. [student1_InsightUBC, student2_InsightUBC])? If so, we just have to tweak this slightly so we can handle this case!